### PR TITLE
Add Layout 2020 support for `display: list-item`

### PR DESCRIPTION
**Pull request by SimonSapin**

### CODE REVIEW

File: components/layout_2020/lists.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        UrlOrNone::Url(url) => Some(vec![
+            PseudoElementContentItem::Replaced(ReplacedContent::from_image_url(
+                info.node, context, url,
+            )?),
+            PseudoElementContentItem::Text(" ".into()),
+        ]),
+        UrlOrNone::None => None,
+    };
+    marker_image().or_else(|| {
+        Some(vec![PseudoElementContentItem::Text(
```
Comment (Manishearth) [inline components/layout_2020/lists.rs:33]: why not just inline this into the above closure?

**Review by Manishearth — APPROVED** [review]

File: components/layout_2020/lists.rs (hunk 1 above)
Comment (SimonSapin) [reply components/layout_2020/lists.rs:33]: The closure uses `?` so it has two paths that can return `None`. This implements “If the `<image>` represents a valid image” in https://drafts.csswg.org/css-lists/#marker-image

**Comment by jdm** [comment]
@bors-servo retry
* #24786

**Pull request merged**